app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -155,7 +129,8 @@ class FirebaseAuthRepository : AuthRepository {
       reauthenticate(
           user,
           password,
-          activity,
+          context,
+          coroutineScope,
           {
             Log.d("DeleteAccount", "User re-authenticated")
```

Comment:
Warning: Is it correct, that the deletion of the account only really happens after the user is reauthenticated? Shouldn't it be that first the account is deleted and then the user gets the option to reauthenticate?

Reply:
It's the first one. For a user to delete its account, he needs to be re-authenticated (to make sure it's him). That's why he needs to re-authenticate before deleting